### ISSUE TITLE
在config文件中，加入了自己的classes，但是推理时标签不是自定义的标签

### ISSUE THREAD

**Issue by Bing-xiao1221** [open-mmlab/mmyolo#877]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmyolo/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmyolo.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmyolo).

### 💬 Describe the reimplementation questions

在自定义数据集上重新实现模型库中的模型的时候，按照步骤在config文件添加了自己的类别后，推理时（运行image_demo.py)输出的类别名不是自定义的类别，而是原本的coco类别。

### Environment

addict                 2.4.0
albumentations         1.3.1
aliyun-python-sdk-core 2.13.36
aliyun-python-sdk-kms  2.16.1
certifi                2023.7.22
cffi                   1.15.1
charset-normalizer     3.2.0
click                  8.1.7
colorama               0.4.6
contourpy              1.1.0
crcmod                 1.7
cryptography           41.0.3
cycler                 0.11.0
fonttools              4.41.0
gitdb                  4.0.10
GitPython              3.1.32
idna                   3.4
imageio                2.31.2
importlib-metadata     6.8.0
importlib-resources    6.0.0
jmespath               0.10.0
joblib                 1.3.2
kiwisolver             1.4.4
lazy_loader            0.3
Markdown               3.4.4
markdown-it-py         3.0.0
matplotlib             3.7.2
mdurl                  0.1.2
mmcv                   2.0.1
mmcv-lite              2.0.1
mmdet                  3.1.0
mmengine               0.8.4
mmyolo                 0.6.0                /home/xuyue/PycharmProjects/mmyolo-main
model-index            0.1.11
networkx               3.1
numpy                  1.25.2
opencv-python          4.8.0.74
opendatalab            0.0.10
openmim                0.3.9
openxlab               0.0.21
ordered-set            4.1.0
oss2                   2.17.0
packaging              23.1
pandas                 2.1.0
Pillow                 10.0.0
pip                    23.2.1
platformdirs           3.10.0
prettytable            3.8.0
psutil                 5.9.5
py-cpuinfo             9.0.0
pycocotools            2.0.7
pycparser              2.21
pycryptodome           3.18.0
Pygments               2.16.1
pyparsing              3.0.9
python-dateutil        2.8.2
pytz                   2023.3
PyWavelets             1.4.1
PyYAML                 6.0.1
qudida                 0.0.4
requests               2.28.2
rich                   13.4.2
scikit-image           0.21.0
scikit-learn           1.3.0
scipy                  1.11.1
seaborn                0.12.2
setuptools             60.2.0
shapely                2.0.1
six                    1.16.0
smmap                  5.0.0
tabulate               0.9.0
termcolor              2.3.0
terminaltables         3.1.10
thop                   0.1.1.post2209072238
threadpoolctl          3.2.0
tifffile               2023.8.25
tomli                  2.0.1
torch                  1.10.0+cu111
torchaudio             0.10.0+rocm4.1
torchvision            0.11.0+cu111
tqdm                   4.65.2
typing_extensions      4.7.1
tzdata                 2023.3
ultralytics            8.0.145
urllib3                1.26.16
wcwidth                0.2.6
wheel                  0.38.4
yapf                   0.40.1
zipp                   3.16.2

**Comment by azad96** [comment]
Hey, I had the same problem as well when using my yolov8 model trained with ultralytics. You probably have a line in your terminal output of your inference as below:
`inference.py:90: UserWarning: dataset_meta or class names are not saved in the checkpoint's meta data, use COCO classes by default.`

What I have found out is the class names are received from the loaded model rather than our custom dataset classes. If this is the case for you, what you should do is to modify the `yolov8_to_mmyolo.py` file. You can also load and modify your model, and then save it if your situation is slightly different. The checkpoint is currently saved as:
~~~
# save checkpoint
checkpoint = dict()
checkpoint['state_dict'] = state_dict
torch.save(checkpoint, dst)
~~~

You just need to adjust it in the following way:
~~~
classes = (
    'label1',
    'label2',
    ...
    'labelX'
)

# save checkpoint
checkpoint = dict()
checkpoint['state_dict'] = state_dict
checkpoint['meta'] = {'CLASSES': classes}
torch.save(checkpoint, dst)
~~~
